client/sources/ok_test/models.py

```diff
@@ -100,9 +100,11 @@ def unlock(self, interact):
                                 for case in suite.cases])
         for suite_num, suite in enumerate(self.suites):
             for case_num, case in enumerate(suite.cases):
+                case_id = '{} > Suite {} > Case {}'.format(
```

Comment:
Well I guess it technically is a concern if I want accurate data, so I'd rather go with your solution that isn't susceptible to students changing their files.

If I understand your proposal correctly, you are suggesting that for a given line of data the prompt is a combination of all the text that came before the student's answer for that case, even when the case has a student submit multiple answers? In addition, if a student answers the prior prompt incorrectly the prompt of the current answer will not be affected by it?

For example:
If a student's session looks like this:
```
>>> x = 5
>>> square(x)
? 36
-- Not quite. Try again! --
? 25
-- OK! --
>>> square(-x)
? 25
-- OK! --
```

The prompt for the first entry would be: `">>> x=5\n>>> square(x)\nhash(25)"`
The prompt for the second entry would be: `">>> x=5\n>>> square(x)\n25>>> square(-x)\nhash(25)"`